build/templates/session.py.mako

```diff
@@ -48,7 +48,25 @@ class ${session_context_manager}(object):
 
 
 % endif
-class _SessionBase(object):
+% for rep_cap in config['repeated_capabilities']:
+class _${rep_cap['python_class_name']}(object):
+    def __init__(self, ${config['session_handle_parameter_name']}, library, encoding):
+        self._${config['session_handle_parameter_name']} = ${config['session_handle_parameter_name']}
+        self._library = library
+        self._encoding = encoding
+
+    def __getitem__(self, repeated_capability):
+        '''Set/get properties or call methods with a repeated capability (i.e. channels)'''
+        if isinstance(repeated_capability, list):
```

Comment:
This seems restrictive. Clients should be able to pass anything iterable even if not a list. For example, range() doesn't return a list.

Reply:
Changed to use duck typing. Anything iterable will now work.